test/integration/issues_test.lua

```diff
@@ -208,3 +209,53 @@ function g.test_twophase_config_locked()
         t.assert_equals(helpers.list_cluster_issues(master), {})
     end)
 end
+
+function g.test_state_hangs()
+    local master = g.cluster.main_server
+    local replica1 = g.cluster:server('replica1')
+
+    t.assert_equals(helpers.list_cluster_issues(master), {})
+    t.assert_equals(helpers.list_cluster_issues(replica1), {})
+
+    local old_timeout = replica1.net_box:eval([[
+        local conf_vars = require('cartridge.vars').new('cartridge.confapplier')
+        local old_timeout = conf_vars.state_notification_timeout
+        conf_vars.state_notification_timeout = 0
+
+        package.loaded['mymodule-permanent'].apply_config = function()
+            -- instance will stop hanging in ConfiguringRoles after 0.3s
+            require('fiber').sleep(0.3)
+        end
+
+        return old_timeout
+    ]])
+
+    fiber.new(function()
+        master:graphql({
+            query = [[mutation($uuids: [String]) {
+                cluster { config_force_reapply(uuids: $uuids) }
+            }]],
+            variables = {uuids = {replica1.instance_uuid}}
+        })
+    end)
+
+    fiber.sleep(0.1)
+    t.assert_items_include(helpers.list_cluster_issues(replica1), {{
+        level = 'warning',
+        topic = 'state_stuck',
+        instance_uuid = replica1.instance_uuid,
+        replicaset_uuid = replica1.replicaset_uuid,
+        message = 'ConfiguringRoles state hangs'..
+            ' on localhost:13302 (replica1)',
+    }})
+    fiber.sleep(0.3)
+    -- When instance passes ConfiguringRoles issue is gone
+    t.assert_items_include(helpers.list_cluster_issues(replica1), {})
+
+    replica1.net_box:eval(string.format([[
+        local conf_vars = require('cartridge.vars').new('cartridge.confapplier')
+        conf_vars.state_notification_timeout = %d
+
+        package.loaded['mymodule-permanent'].apply_config = nil
+    ]], old_timeout))
```

Comment:
Won't it work?

```suggestion
    replica1.net_box:eval([[
        local conf_vars = require('cartridge.vars').new('cartridge.confapplier')
        conf_vars.state_notification_timeout = ...

        package.loaded['mymodule-permanent'].apply_config = nil
    ]], old_timeout))
```

Reply:
```suggestion
    replica1.net_box:eval([[
        local conf_vars = require('cartridge.vars').new('cartridge.confapplier')
        conf_vars.state_notification_timeout = ...

        package.loaded['mymodule-permanent'].apply_config = nil
    ]], {old_timeout})
```

Reply:
Removed extra parenthesis.
It will, yeah. Much obliged!